### ISSUE TITLE
SignalXY: final segment is not being rendered

### ISSUE THREAD

**Issue by mjazd** [ScottPlot/ScottPlot#3423]
**Issue:** Some lines are not being renderend depending on the size of the image. Attached is one example, the issue also occurs at different image sizes and for different lines.

**ScottPlot Version:** ScottPlot v5.0.21

**Code Sample:**

```cs
var plot = new Plot();

plot.Axes.Left.MaximumSize = 40.5f;
plot.Axes.Left.MinimumSize = 40.5f;
plot.Axes.Right.MaximumSize = 19.5f;
plot.Axes.Right.MinimumSize = 19.5f;

plot.Add.SignalXY(
    new double[] { 1284, 1485 },
    new double[] { -2.373134328358208, -2.373134328358208 }
);

plot.Axes.SetLimits(0, 3801.1111111111113, -7.092537313432836, -0.7999999999999993);

plot.SavePng("682.png", 682, 186); // line not rendered
plot.SavePng("683.png", 683, 186); // line rendered correctly
```
682.png
![682](https://github.com/ScottPlot/ScottPlot/assets/161584436/c652a3b3-ec7f-480b-9cfe-c1d8bd18079b)

683.png
![683](https://github.com/ScottPlot/ScottPlot/assets/161584436/2da05b6d-9b1a-42b3-a6f4-886415286c19)

**Comment by swharden** [comment]
Thanks for reporting this!

A related issue is that the last segment of SignalXY plots doesn't seem to be rendered...

```cs
using ScottPlot;

namespace Sandbox.WinForms;

public partial class Form1 : Form
{
    public Form1()
    {
        InitializeComponent();

        var line = formsPlot1.Plot.Add.Line(0, 0, 1, 0);
        line.Label = "Line";
        line.LineWidth = 5;

        var signal = formsPlot1.Plot.Add.Signal(new double[] { 1, 1 });
        signal.Label = "Signal";
        signal.LineWidth = 5;

        double[] xs = Generate.Consecutive(3);
        double[] ys = Generate.Repeating(3, 2);
        var signalXY = formsPlot1.Plot.Add.SignalXY(xs, ys);
        signalXY.Label = "SignalXY";
        signalXY.LineWidth = 5;

        formsPlot1.Plot.ShowLegend();
        formsPlot1.Plot.HideGrid();
    }
}
```

![image](https://github.com/ScottPlot/ScottPlot/assets/4165489/a2f76423-a697-496e-854c-a9613beca97c)

**Comment by mjazd** [comment]
The issue I've originally reported wasn't solved by #3495. The code sample from the first message still provides incorrect results as of version 5.0.23.
